manifest.json

```diff
@@ -41,5 +41,13 @@
     "projectOptions": ["app", "addon"],
     "nodeVersion": "6.0.0",
     "commands": ["jscodeshift -t https://raw.githubusercontent.com/simplabs/qunit-dom-codemod/master/qunit-dom-codemod.js ./tests"]
+  },
+  "ember-no-implicit-this-codemod": {
+    "versions": {
+      "ember-source": "3.12.0"
+    },
+    "projectOptions": ["app", "addon"],
+    "nodeVersion": "8.0.0",
+    "script": "const cp = require('child_process'); let ps = cp.spawn('ember', ['s']); let url = await new Promise(resolve => { ps.stdout.on('data', data => { let str = data.toString(); let matches = str.match(/^Build successful \\(\\d+ms\\) – Serving on (.*)$/m); if (matches) { resolve(matches[1]); } }); }); cp.execSync(`npx ember-no-implicit-this-codemod ${url} app/`); ps.kill(); await new Promise(resolve => { ps.on('exit', resolve); });"
```

Comment:
this seems risky :-\

Reply:
elaborate please

Reply:
just that having a bunch of node stored as a string implies that this'll be eval'd somewhere, and I know that eval can lead to vulnerabilities.

Makes me wonder if the codemods should distribute interactive bins that ember-cli-update invokes -- that way each codemod could be responsible for instructing the user how to set things up. idk

Reply:
I did this to help protect https://github.com/kellyselden/boilerplate-update/pull/125

But in the long run, there isn't too much difference between eval and npx, both download code and execute.

Reply:
Well, in this case it is embedding `eval` **and** `npx` 😸. Two semi-risky things seems strictly worse than one no?

Reply:
I'm torn. I know there is risk associated with `eval`, but I don't think there is much additional risk here. My thoughts:

* The dangers of `eval` are mostly for people concatenating strings from user input, much like SQL injection. The strings in this case are under our control and don't take user input.
* I use the `Function` contructor instead of `eval` described [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval#Do_not_ever_use_eval!).
* So the attack vector seems to be a compromised GitHub account, which seems as likely as a compromised NPM account.

Please correct me if I'm wrong.